apps/frontend/src/pages/plan-generation/plan-generation.tsx

```diff
@@ -0,0 +1,135 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+import {
+	Book,
+	Croissant,
+	CupCoffee,
+	CupGreen,
+	Drink,
+	Greens,
+	IceCream,
+	Laptop,
+	OrangeWhole,
+	Phone,
+	Teddy,
+} from "~/assets/img/shared/illustrations/illustrations.img.js";
+import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
+import { getClassNames } from "~/libs/helpers/get-class-names.js";
+import { useAppDispatch, useAppSelector } from "~/libs/hooks/hooks.js";
+import { StorageKey } from "~/libs/modules/storage/storage.js";
+import { actions as planActions } from "~/modules/plans/plans.js";
+
+import { ImageSlider } from "./components/slider/slider.js";
+import {
+	FAST_INCREMENT,
+	FAST_INTERVAL_MS,
+	PROGRESS_MAX,
+	PROGRESS_MAX_SLOW,
+	PROGRESS_MIN,
+	SLOW_INCREMENT_DIVISOR,
+	SLOW_INCREMENT_MAX,
+	SLOW_INTERVAL_MS,
+} from "./libs/constants/constants.js";
+import { type QuizAnswersDto } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+const slides = [
+	Teddy,
+	Book,
+	Croissant,
+	CupCoffee,
+	CupGreen,
+	Drink,
+	Greens,
+	IceCream,
+	Laptop,
+	OrangeWhole,
+	Phone,
+];
+
+const PlanGeneration: React.FC = () => {
+	const [progress, setProgress] = useState<number>(PROGRESS_MIN);
+
+	const dispatch = useAppDispatch();
+	const status = useAppSelector((state) => state.plan.dataStatus);
+
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		const stored = localStorage.getItem(StorageKey.QUIZ_ANSWER);
+		const quizAnswers: QuizAnswersDto = stored
+			? (JSON.parse(stored) as QuizAnswersDto)
+			: [];
+
+		void dispatch(planActions.generate(quizAnswers));
+
+		const slowProgressId = setInterval(() => {
+			setProgress((previous) => {
```

Comment:
I would also move the whole callback you pass to the setProgress function out of the component and would name it `getUpdatedProgress` or something like that